components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1250,14 +1236,43 @@ private int getMaximumItemPerPage() {
         return maximumItemsPerPage;
     }
 
+    /**
+     * Get the Default Items per Page needed to display.
+     *
+     * @return defaultItemsPerPage need to display.
+     */
+    private int getDefaultItemsPerPage() {
+
+        int defaultItemsPerPage = -1;
+        try {
+            String defaultItemsPerPageProperty = IdentityUtil.getProperty(IdPManagementConstants
+                    .DEFAULT_ITEMS_PRE_PAGE_PROPERTY);
+            if (StringUtils.isNotBlank(defaultItemsPerPageProperty)) {
+                defaultItemsPerPage = Integer.parseInt(defaultItemsPerPageProperty);
```

Comment:
if the parsed value is greater than 0, assign it to `defaultItemsPerPage` variable.